docs/DeveloperGuide.md

```diff
@@ -1,38 +1,619 @@
-# Developer Guide
+# Apollo Developer Guide
+
+## Table of Contents
+
+1. [Acknowledgments](#acknowledgements)
+2. [Design](#design)
+    + [Architecture](#architecture)
+    + [UI Component](#ui-component)
+    + [Logic Component](#logic-component)
+    + [Model Component](#model-component)
+    + [Storage Component](#storage-component)
+    + [Common Classes](#common-classes)
+3. [Implementation](#implementation)
+    + [Add Module](#add-module)
+    + [Delete Module](#delete-module)
+    + [List Modules](#list-modules)
+    + [Show Module](#show-module)
+    + [Add Task](#add-task)
+    + [Delete Task](#delete-task)
+    + [Mark Task As Done](#mark-task-as-done)
+    + [Unmark Task](#unmark-task)
+    + [Find Task](#find-task)
+    + [List Task](#list-task)
+    + [Find Task on Date](#find-task-on-date)
+    + [Storage](#storage)
+    + [Logging](#logging)
+4. [Documentation, logging, testing, configuration, dev-ops](#documentations-logging-testing-configuration-dev-ops)
+    + [Documentation](#documentation)
+    + [Logging](#logging)
+    + [Testing](#testing)
+    + [Configuration](#configuration)
+    + [Dev-ops](#dev-ops)
+5. [Appendix A: Requirements](#appendix-a-requirements)
+    + [Product Scope](#product-scope)
+        + [Target User Profile](#target-user-profile)
+        + [Value Proposition](#value-proposition)
+    + [User Stories](#user-stories)
+    + [Use Cases](#use-cases)
+    + [Non-Functional Requirements](#non-functional-requirements)
+    + [Glossary](#glossary)
+6. [Appendix B: Instructions for manual testing](#appendix-b-instructions-for-manual-testing)
+    + [Launch and Shutdown](#launch-and-shutdown)
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+We would like to acknowledge:
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
```

Comment:
There is a sequence diagram in the folder which is clear and good. However it is not added in the developer guide so it's not rendered, you might wish to add it in. It will be good to have more class diagrams/sequence diagrams to help with the illustration.